lib/molinillo/resolution.rb

```diff
@@ -248,6 +262,30 @@ def state_index_for_unwind
         index
       end
 
+      # @param [String] name of a dependency
+      # @return [Object] the requirement that led to a version of a possibility
+      #   with the given name being activated.
+      def initial_requirement_for_name(name)
+        return nil unless activated.vertex_named(name)
+        return nil unless activated.vertex_named(name).payload
+        states.find { |s| s.name == name }.requirement
+      end
+
+      # @param [Conflict] conflict
+      # @return [Array] minimal array of requirements that would cause the passed
+      #    conflict to occur.
+      def binding_requirements_for_conflict(conflict)
+        return [conflict.requirement] unless conflict.possibility_set && conflict.possibility_set.possibilities.any?
+
+        conflict.requirements.values.flatten(1).uniq.each do |req|
```

Comment:
store `conflict.requirements.values.flatten(1).uniq` in a variable, since we're using it twice?